add hybrid programming solutions chapter (#9)
Bayer-Group/sas2r: 13-hybrid-programming.Rmd

# Hybrid Programming with SAS and R

This section contains current solutions of hybrid programming.

## Use {sasr}

{[sasr](https://github.com/insightsengineering/sasr)} is a package built on {reticulate} so that the python package {[saspy](https://github.com/sassoftware/saspy)} can be called from R.
It adds some simple wrappers to the {saspy} functionalities to make the syntax similar to R.

Here are some simple examples of how you can use it.
```{r eval = FALSE}
library(sasr)

> # transfer data from R to SAS
> df2sd(mtcars)
> 
> # execute SAS commands
> 
> run_sas("
>   data test;
>     set work.mtcars;
>   run;
> ")
> 
> # transfer data back to R
> 
> sd2df("test")
> 

Reviewer: ```suggestion
# Transfer data from R to SAS.
df2sd(mtcars)

# Execute SAS commands.
run_sas("
  data test;
    set work.mtcars;
  run;
")

# Transfer data back to R.
sd2df("test")
```

```

> This solution is based on {[saspy](https://github.com/sassoftware/saspy)}, an official python interface to connect SAS.

Reviewer: this is partly a repetition from above? maybe already mention this above instead?

> This requires that you have a machine with SAS and it accessible through specific [connection types](https://sassoftware.github.io/saspy/configuration.html).

Reviewer: maybe mention here the SSH example since it will be common

## Use {jupyter}/{quarto} and {saspy}

[jupyter](http://www.jupyter.org/) and [quarto](https://quarto.org/) are both capable of creating dynamic outputs from multiple languages.

> Although official SAS kernel is not supported, it can be installed through package [sas_kernel](https://github.com/sassoftware/sas_kernel) provided by [SAS software](https://github.com/sassoftware).

Reviewer: ```suggestion
Although the official SAS kernel is not supported, it can be installed through package [sas_kernel](https://github.com/sassoftware/sas_kernel) provided by [SAS software](https://github.com/sassoftware).
```

A quick example can be found at [Harvey Lieberman's blog](https://www.harveyl888.com/post/2022-06-27-quarto_sas_01/).

The requirement is the same as {sasr} as they both use {saspy} as the backend.

## Use {SASmarkdown}

> {[SASmarkdown](https://github.com/Hemken/SASmarkdown)} is a pacakge that installs SAS engine to {knitr} so that in your .Rmd files

Reviewer: ```suggestion
{[SASmarkdown](https://github.com/Hemken/SASmarkdown)} is a package that installs the SAS engine to {knitr} so that in your `.Rmd` files
```

using SAS engines is possible.

> However, this requires that SAS is installed on your machine.

Reviewer: ```suggestion
However, this requires that SAS is installed locally on your machine.
```

Reviewer 2: ... in contrast to client to a server?
